app/src/main/java/org/thoughtcrime/securesms/service/KeyCachingService.java

```diff
@@ -252,11 +253,18 @@ private void foregroundService() {
     builder.setContentIntent(buildLaunchIntent());
 
     stopForeground(true);
-    if (Build.VERSION.SDK_INT >= 34) {
-      startForeground(SERVICE_RUNNING_ID, builder.build(), ServiceInfo.FOREGROUND_SERVICE_TYPE_SPECIAL_USE);
-    } else {
-      startForeground(SERVICE_RUNNING_ID, builder.build());
+
+    int type = 0;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
```

Comment:
Some would say the int literal is easier to read, e.g. I haven't memorised what version `UPSIDE_DOWN_CAKE` is, but I know 34 or so is our target, and 23 is our lowest version, but I don't know the name of that version, and these values are truly constant, they're constant inside AND outside the app, they will never change.

```suggestion
    if (Build.VERSION.SDK_INT >= 34) {
```